attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -56,6 +56,26 @@ async fn get_request_extra_params() -> serde_json::Value {
     extra_params
 }
 
+async fn get_hash_algorithm(extra_params: serde_json::Value) -> Result<HashAlgorithm> {
```

Comment:
we might not need this to be `async`

Reply:
@Xynnn007 - true, this function doesn't actually _need_ to be async, but I feel the code is cleaner + safer if we use async everywhere we can reasonably do so.

If I drop the `async`, I either:

- block the calling task implicitly (maybe not an issue now, but potentially problematic if the implementation changes (future bug potential)).
- block it explicitly with `spawn_blocking()` (extra code noise for no obvious benefit).

Reply:
My personal preference is to mark it as non-async. Once it needs to be async, we can then mark it as async version.

btw, if a function is a concrete implementation of an async-trait function, I can live with situations where even no async code is inside the function body.

wdyt?

Reply:
I think we've spent enough time discussing a 5 byte change. Branch updated :smile: